langroid/agent/special/sql/sql_chat_agent.py

```diff
@@ -117,56 +115,92 @@ class SQLChatAgentConfig(ChatAgentConfig):
+        self._init_message_tools()
+
+    def _validate_config(self, config: "SQLChatAgentConfig") -> None:
+        """Validate the configuration to ensure all necessary fields are present."""
+        if config.database_session is None and config.database_uri is None:
+            raise ValueError("Database information must be provided")
+
+    def _init_database(self) -> None:
+        """Initialize the database engine and session."""
+        if self.config.database_session:
+            self.Session = self.config.database_session
             self.engine = self.Session.bind
         else:
-            self.engine = create_engine(config.database_uri)
+            self.engine = create_engine(self.config.database_uri)
             self.Session = sessionmaker(bind=self.engine)()
-        self.metadata = MetaData()
 
+    def _init_metadata(self) -> None:
+        """Initialize the database metadata."""
         if self.engine is None:
             raise ValueError("Database engine is None")
-        self.metadata.reflect(self.engine)
 
+        self.metadata = MetaData()
+        self.metadata.reflect(self.engine)
         logger.info(
-            f"""SQLChatAgent initialized with database: 
-            {self.engine} and tables: 
-            {self.metadata.tables}
-            """
+            "SQLChatAgent initialized with database: %s and tables: %s",
+            self.engine,
+            self.metadata.tables,
         )
 
-        if not config.context_descriptions and isinstance(self.engine, Engine):
-            config.context_descriptions = extract_schema_descriptions(self.engine)
+    def _init_table_metadata(self) -> None:
+        """Initialize metadata for the tables present in the database."""
+        if not self.config.context_descriptions and isinstance(self.engine, Engine):
+            self.config.context_descriptions = extract_schema_descriptions(self.engine)
 
-        # Combine database information with context descriptions
-        schema_dict = combine_metadata(self.metadata, config.context_descriptions)
+        if self.config.use_schema_tools:
+            self.table_metadata = populate_metadata_with_schema_tools(
+                self.metadata, self.config.context_descriptions
+            )
+        else:
+            self.table_metadata = populate_metadata(
+                self.metadata, self.config.context_descriptions
+            )
 
-        # Update the system message with the table information
-        self.config.system_message = self.config.system_message.format(
-            schema_dict=schema_dict, dialect=self.engine.dialect.name
-        )
+    def _init_message_tools(self) -> None:
+        """Initialize message tools used for chatting."""
+        message = self._format_message()
+        self.config.system_message = self.config.system_message.format(mode=message)
+        super().__init__(self.config)
+        self.enable_message(RunQueryTool)
+        if self.config.use_schema_tools:
```

Comment:
@pchalasani this is where use_schema_tools is being used

Reply:
it says outdated , so it looks like this statement was in an older version, but not in latest... see pic

<img width="692" alt="image" src="https://github.com/langroid/langroid/assets/554347/1f364c21-ca86-4e75-b775-e5c281c824ce">